cmd/web-handlers.go

```diff
@@ -1037,6 +1044,8 @@ func toWebAPIError(err error) APIError {
 		apiErrCode = ErrReadQuorum
 	case PolicyNesting:
 		apiErrCode = ErrPolicyNesting
+	case NotImplemented:
+		apiErrCode = ErrNotImplemented
```

Comment:
I had suggested to use ErrNotImplemented, but @poornas you can use this as it gives more control on the error message:

```
krishna@escape:~/gocode/src/github.com/minio/minio$ git diff
diff --git a/cmd/web-handlers.go b/cmd/web-handlers.go
index bc4992e..29d25d8 100644
--- a/cmd/web-handlers.go
+++ b/cmd/web-handlers.go
@@ -1037,6 +1037,12 @@ func toWebAPIError(err error) APIError {
                apiErrCode = ErrReadQuorum
        case PolicyNesting:
                apiErrCode = ErrPolicyNesting
+       case NotImplemented:
+               return APIError{
+                       Code:           "NotImplemented",
+                       HTTPStatusCode: http.StatusBadRequest,
+                       Description:    "Functionality not implemented",
+               }
        default:
                // Log unexpected and unhandled errors.
                errorIf(err, errUnexpected.Error())
krishna@escape:~/gocode/src/github.com/minio/minio$ 
```

But now the code looks inconsistent as we return only in one case statement. So you can do this:
```
        // Convert error type to api error code.
        var apiErrCode APIErrorCode
        switch err.(type) {
        case StorageFull:
                return getAPIError(ErrStorageFull)
        case BucketNotFound:
                return getAPIError(ErrNoSuchBucket)
        case BucketExists:
                return getAPIError(ErrBucketAlreadyOwnedByYou)
        case BucketNameInvalid:
                return getAPIError(ErrInvalidBucketName)
        case BadDigest:
                return getAPIError(ErrBadDigest)
        case IncompleteBody:
                return getAPIError(ErrIncompleteBody)
        case ObjectExistsAsDirectory:
                return getAPIError(ErrObjectExistsAsDirectory)
        case ObjectNotFound:
                return getAPIError(ErrNoSuchKey)
        case ObjectNameInvalid:
                return getAPIError(ErrNoSuchKey)
        case InsufficientWriteQuorum:
                return getAPIError(ErrWriteQuorum)
        case InsufficientReadQuorum:
                return getAPIError(ErrReadQuorum)
        case PolicyNesting:
                return getAPIError(ErrPolicyNesting)
        case NotImplemented:
                return APIError{
                        Code:           "NotImplemented",
                        HTTPStatusCode: http.StatusBadRequest,
                        Description:    "Functionality not implemented",
                }
        }
        // Log unexpected and unhandled errors.
        errorIf(err, errUnexpected.Error())
        return APIError{
                Code:           "InternalError",
                HTTPStatusCode: http.StatusInternalServerError,
                Description:    err.Error(),
        }
```